backend/app/services/llm_service.py

```
@@ -1,242 +1,182 @@
+        )
+    
+    if user_data.organization_id:
+        organization_result = await db.execute( select(Organization).where(Organization.id == user_data.organization_id))
+        organization = organization_result.scalar_one_or_none()
+        if not organization:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail="Organization not found"
+            )
+    
+    if user_data.role == UserRole.USER and user.role == UserRole.ADMIN or user_data.is_active == False and user.role == UserRole.ADMIN:
```

Ruff (0.15.13)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)